### PR TITLE
feat: set up basic project structure

### PR DESCRIPTION
**Pull request by ComradeProgrammer**
feat: set up basic project structure

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@
 
 # Dependency directories (remove the comment below to include it)
 # vendor/
+k8s-authz
```
Comment (hsluoyz) [inline .gitignore]: Don't commit this

File: hack/boilerplate.go.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2022 The Casbin Authors. All Rights Reserved.
```
Comment (hsluoyz) [inline hack/boilerplate.go.txt:1]: What's this file used for?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@sagilio plz review

File: hack/boilerplate.go.txt (hunk 2 above)
Comment (ComradeProgrammer) [reply hack/boilerplate.go.txt:1]: There will be codes generated by k8s official tools, and this file is used to stipulate the headers we want to insert in those generated files

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**